### PR TITLE
scrub: do not assign value if read error

### PR DESCRIPTION
**Pull request by dragonylffly**
Signed-off-by: Li Wang <li.wang@kylin-cloud.com>

### CODE REVIEW

File: src/osd/ReplicatedBackend.cc [hunk 1]
```diff
@@ -778,9 +778,10 @@ void ReplicatedBackend::be_deep_scrub(
     dout(25) << __func__ << "  " << poid << " got "
 	     << r << " on read, read_error" << dendl;
     o.read_error = true;
```
Comment (dzafman) [inline src/osd/ReplicatedBackend.cc]: Maybe we could just return in the error case.

File: src/osd/ReplicatedBackend.cc [hunk 2]
```diff
@@ -832,11 +833,11 @@ void ReplicatedBackend::be_deep_scrub(
     dout(25) << __func__ << "  " << poid << " got "
 	     << r << " on omap scan, read_error" << dendl;
     o.read_error = true;
-  }
-
+  } else {
```
Comment (dzafman) [inline src/osd/ReplicatedBackend.cc]: An omap_get_header() EIO could still get here if omap_get_iterator() doesn't error.  A return whenever setting read_error would handle the 3 error cases here.

**Comment by dzafman** [comment]
Would need the equivalent change in ECBackend::be_deep_scrub()

**Comment by dragonylffly** [comment]
@dzafman thanks for reviewing, updated

**Comment by dzafman** [comment]
Reviewed-by once testing passes.

Come to think of it.  Is this really a bug fix?  Maybe this should be marked "cleanup."

**Pull request merged**